lib/fluent/plugin/out_forward.rb

```diff
@@ -144,6 +147,13 @@ def configure(conf)
         log.info "adding forwarding server '#{name}'", host: host, port: port, weight: weight, plugin_id: plugin_id
       }
 
+      if @compress == :gzip && @buffer.compress == :text
+        @buffer.compress = :gzip
+      elsif @compress == :text && @buffer.compress == :gzip
+        @compress = :gzip
```

Comment:
IMO there are some needs to compress buffer chunks and not to compress network traffic, for who want just to reduce disk usage.
So enabling `@compress` forcedly is too much for such users. It's just needed to notice users that this configuration (buffer compress, network traffic uncompressed) consumes more CPU time.